File: 28-implement-strstr/28-implement-strstr.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
    
   // console.log(haystack.substring(i,needle.length))
    if(needle.length == 0){
        return 0;
    }
    for(var i = 0; i<haystack.length;i++){
        if(haystack.substring(i,needle.length+i) == needle){
            return i;
        }
       
    }
     return -1;
};

File: 28-implement-strstr/28-implement-strstr.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        if(needle.size()==0) return 0;
        for(int i=0,j=needle.size()-1;j<haystack.size();i++,j++){
            if(haystack.substr(i,needle.size())==needle)
                return i;
        }
        return -1;
    }
};